src/app.js

```diff
@@ -269,43 +289,56 @@ var run = function () {
     var name = $(this).parent().find('.name').text()
 
     if (confirm('Are you sure you want to remove: ' + name + ' from local storage?')) {
-      storage.remove(name)
-      editor.removeSession(name)
-      editor.setNextFile(name)
-      updateFiles()
+      files.remove(name)
+      switchToNextFile()
+      editor.discard(name)
     }
     return false
   })
 
   editor.event.register('sessionSwitched', updateFiles)
 
   function switchToFile (file) {
-    editor.setCacheFile(file)
-    updateFiles()
+    currentFile = file
+
+    if (files.isReadOnly(file)) {
+      editor.openReadOnly(file, files.get(file))
+    } else {
+      editor.open(file, files.get(file))
+    }
   }
 
+  function switchToNextFile () {
+    switchToFile(Object.keys(files.list())[0])
```

Comment:
I checked browser test failure. It might fails because of that. 
Are we sure that the array is always not empty?
and I don't get why this will switch to the next file (is that because the previous files has been "poped" ?